compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -602,6 +603,20 @@ public void visit(BLangSimpleVariable varNode) {
                 !types.checkListenerCompatibility(varNode.symbol.type)) {
             dlog.error(varNode.pos, DiagnosticCode.INVALID_LISTENER_VARIABLE, varNode.name);
         }
+
+        transferForkFlag(varNode);
+    }
+
+    private void transferForkFlag(BLangSimpleVariable varNode) {
+        // Transfer FORK flag to workers future value.
+        if (varNode.expr != null && varNode.expr.getKind() == NodeKind.INVOCATION
+                && varNode.flagSet.contains(Flag.WORKER)) {
+
+            BLangInvocation expr = (BLangInvocation) varNode.expr;
+            if (expr.name.value.startsWith("0") && (expr.symbol.flags & Flags.FORKED) == Flags.FORKED) {
```

Comment:
This validation is not scalable. let's create an issue to track this and clean up the implementation.

Reply:
https://github.com/ballerina-platform/ballerina-lang/issues/18680 created to track this